crates/bevy_ecs/src/world/mod.rs

```diff
@@ -591,64 +591,104 @@ impl World {
         }
     }
 
+    /// Inserts a new resource with default values.
+    ///
+    /// If the resource already exists, nothing happens.
+    ///
+    /// Uses the [`FromWorld`] trait to determine values,
+    /// which has a blanket impl for all `T: Default`.
```

Comment:
```suggestion
    /// which has a blanket impl for all `R: Default`.
```
For me, this would fit better since we use R as the generic. If you agree, please do the same change for `init_non_send_resource`.

Reply:
Ah, I missed this one!